src/Common/ExecTrait.php

```diff
@@ -215,12 +215,29 @@ public function envVars(array $env)
      *
      * @return $this
      */
-    public function setInput($input)
+    public function setProcessInput($input)
     {
         $this->input = $input;
+        // A tty should not be allocated when the input is provided.
+        $this->interactive(false);
         return $this;
     }
 
+    /**
+     * Pass an input to the process. Can be resource created with fopen() or string
+     *
+     * @param resource|string $input
+     *
+     * @return $this
+     *
+     * @deprecated
+     */
+    public function setInput($input)
+    {
+        trigger_error('setInput() is deprecated. Please use setProcessInput(().', E_USER_DEPRECATED);
+        return $this->setProcessInput($input);
```

Comment:
`setInput` should maintain old behavior and not set interactive `false`

Reply:
This is probably not necessary but I can't prove that it won't cause a problem for someone.

Reply:
But you are right, better keep the same behavior for the deprecated method.